tests/unit/test_tokenizer.py

```
@@ -0,0 +1,445 @@
+
+        assert tokenizer.eos_token_id is not None
+        assert tokenizer.eos_token_id == tokenizer.vocab["<|endoftext|>"]
+
+    def test_retrain_clears_state(self):
+        """Test that calling train() multiple times clears previous state."""
+        tokenizer = BPETokenizer(vocab_size=300)
+
+        # First training
+        tokenizer.train(["Hello world test"], verbose=False)
+        first_vocab_size = len(tokenizer.vocab)
+        first_merges = len(tokenizer.merges)
```

Ruff (0.14.2)

Local variable `first_merges` is assigned to but never used

Remove assignment to unused variable `first_merges`

(F841)